From 44f5cb9d04a4c8524731e2b0656b34760c44bd6e Mon Sep 17 00:00:00 2001
From: John Jacques <john.jacques@intel.com>
Date: Wed, 24 Jun 2015 10:05:40 -0500
Subject: [PATCH 36/86] arch/arm64: Axxia Device Tree Updates

Correct the addresses and interrupt numbers for AXC6700.

Signed-off-by: John Jacques <john.jacques@intel.com>
[Original patch is from linux-yocto-4.1 axxia/base branch.]
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 arch/arm64/boot/dts/intel/axc6704-emu.dts |   18 +++++++++++++++-
 arch/arm64/boot/dts/intel/axc67xx.dtsi    |   31 +++++++++++++++--------------
 2 files changed, 33 insertions(+), 16 deletions(-)

diff --git a/arch/arm64/boot/dts/intel/axc6704-emu.dts b/arch/arm64/boot/dts/intel/axc6704-emu.dts
index 71092d7..7be1a65 100644
--- a/arch/arm64/boot/dts/intel/axc6704-emu.dts
+++ b/arch/arm64/boot/dts/intel/axc6704-emu.dts
@@ -12,7 +12,7 @@
 /dts-v1/;
 
 #include "axc67xx.dtsi"
-#include "axc6704-cpus.dtsi"
+#include "axc6701-cpus.dtsi"
 
 / {
 	model = "Lionfish Emulation";
@@ -30,3 +30,19 @@
 &serial0 {
 	status = "okay";
 };
+
+&nemac {
+       status = "okay";
+       phy-handle = <&phy0>;
+};
+
+&mdio0 {
+      status = "okay";
+      lsi,mdio-clk-offset = <0x5>;
+      lsi,mdio-clk-period = <0xc>;
+      max-speed = <10>;
+
+      phy0: ethernet-phy@3 {
+           reg = <0x3>;
+      };
+};
diff --git a/arch/arm64/boot/dts/intel/axc67xx.dtsi b/arch/arm64/boot/dts/intel/axc67xx.dtsi
index e8d3f28..ee7f961 100644
--- a/arch/arm64/boot/dts/intel/axc67xx.dtsi
+++ b/arch/arm64/boot/dts/intel/axc67xx.dtsi
@@ -95,13 +95,14 @@
 
 		syscon: syscon@8002000000 {
 			compatible = "lsi,axxia-syscon", "syscon";
-			reg = <0x80 0x02000000 0x40000>;
+			reg = <0x80 0x02c00000 0x40000>;
 		};
 
 		nemac: ethernet@8080240000 {
-			compatible = "lsi,nemac";
-			reg = <0x80 0x80240000 0x1000>;
-			interrupts = <GIC_SPI 4 IRQ_TYPE_LEVEL_HIGH>;
+			compatible = "intel,nemac";
+			reg = <0x80 0x80240000 0x10000>;
+			interrupts = <GIC_SPI 0 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 1 IRQ_TYPE_LEVEL_HIGH>;
 			status = "disabled";
 		};
 
@@ -350,7 +351,7 @@
 			serial0: uart@8080000000 {
 				compatible = "arm,pl011", "arm,primecell";
 				reg = <0x80 0x80000000 0x10000>;
-				interrupts = <GIC_SPI 71 IRQ_TYPE_LEVEL_HIGH>;
+				interrupts = <GIC_SPI 39 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&clks AXXIA_CLK_PER>;
 				clock-names = "apb_pclk";
 				status = "disabled";
@@ -359,7 +360,7 @@
 			serial1: uart@8080010000 {
 				compatible = "arm,pl011", "arm,primecell";
 				reg = <0x80 0x80010000 0x10000>;
-				interrupts = <GIC_SPI 72 IRQ_TYPE_LEVEL_HIGH>;
+				interrupts = <GIC_SPI 40 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&clks AXXIA_CLK_PER>;
 				clock-names = "apb_pclk";
 				status = "disabled";
@@ -368,7 +369,7 @@
 			serial2: uart@8080020000 {
 				compatible = "arm,pl011", "arm,primecell";
 				reg = <0x80 0x80020000 0x10000>;
-				interrupts = <GIC_SPI 73 IRQ_TYPE_LEVEL_HIGH>;
+				interrupts = <GIC_SPI 41 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&clks AXXIA_CLK_PER>;
 				clock-names = "apb_pclk";
 				status = "disabled";
@@ -377,14 +378,14 @@
 			timer0: timer@8080220000 {
 				compatible = "arm,sp804", "arm,primecell";
 				reg = <0x80 0x80220000 0x10000>;
-				interrupts = <GIC_SPI 62 IRQ_TYPE_LEVEL_HIGH>,
-					     <GIC_SPI 63 IRQ_TYPE_LEVEL_HIGH>,
-					     <GIC_SPI 64 IRQ_TYPE_LEVEL_HIGH>,
-					     <GIC_SPI 65 IRQ_TYPE_LEVEL_HIGH>,
-					     <GIC_SPI 66 IRQ_TYPE_LEVEL_HIGH>,
-					     <GIC_SPI 67 IRQ_TYPE_LEVEL_HIGH>,
-					     <GIC_SPI 68 IRQ_TYPE_LEVEL_HIGH>,
-					     <GIC_SPI 69 IRQ_TYPE_LEVEL_HIGH>;
+				interrupts = <GIC_SPI 30 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 31 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 32 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 33 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 34 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 35 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 36 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 37 IRQ_TYPE_LEVEL_HIGH>;
 				clocks = <&clks AXXIA_CLK_PER>;
 				clock-names = "apb_pclk";
 			};
-- 
1.7.5.4

